File: vertexCover.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
#define N 10000
vector<int> adjlist[N];
int main()
{
  int m,n,k;
  scanf("%d%d",&n,&m);
  for (int i= 0; i<m; i++) //输入边
  {
    int e1,e2;
    scanf("%d%d",&e1,&e2);
    adjlist[e1].push_back(e2);
    adjlist[e2].push_back(e1);
  }
  scanf("%d",&k);//输入查询的次数
  for(int i=0;i<k;++i)
  {
    int nv;//输入本次查询的节点个数
    scanf("%d",&nv);
    vector<int> vertexs,visited;
    for(int j=0;j<nv;j++)
    {
      int temp;
      scanf("%d",&temp);
      vertexs.push_back(temp);//存储需要查询的节点
    }
    int count=0;
    for(int j=0;j<nv;j++)//依次取出要查询的节点
    {
      for(int t=0;t<adjlist[vertexs[j]].size();t++)//检查该节点所连接的边
      {
          if( find(visited.begin(),visited.end(),adjlist[vertexs[j]][t]) != visited.end() )
            count++;
      }
      visited.push_back(vertexs[j]);
    }
    printf("count=%d\n",count);
    printf("%s\n",(count<m?"No":"Yes"));
  }
}

File: vertexCover.py
# -*- coding:utf-8 -*-
def vertexCover():
    n,m=raw_input().split(" ")
    n=int(n);m=int(m)
    adjlist=[[] for i in range(n)]
    # print adjlist
    for i in range(m):
        e1,e2=raw_input().split(" ")
        adjlist[int(e1)].append(int(e2))
        adjlist[int(e2)].append(int(e1))
    k=input()#number of loop
    for i in range(k):
        nv=input()#number of vertexs
        vertexs=[];visited=[]
        vertexs=map(int,raw_input().split(' '))#store vertex
        count=0
        for e in vertexs:#
            for end in adjlist[e]:#
                if end not in visited:
                    count+=1
            visited.append(e)
        print ('No' if count<m else 'Yes')
vertexCover()
